src/visualization/index.ts

```diff
@@ -0,0 +1,46 @@
+import {FunctionComponent, ComponentClass} from 'react'
+import {ViewProperties, Theme, TimeZone, TimeRange} from 'src/types'
+import {FromFluxResult} from '@influxdata/giraffe'
+
+export interface VisOptionProps {
+  properties: ViewProperties
+  results: FromFluxResult
+  update: (obj: any) => void
+}
+
+export interface VisProps {
+  properties: ViewProperties
+  result: FromFluxResult
+  theme?: Theme
+  timeZone?: TimeZone
+  timeRange?: TimeRange
+}
+
+export interface VisTypeRegistration {
+  type: string // a unique string that identifies a visualization
+  name: string // the name that shows up in the dropdown
+  graphic: JSX.Element // the icon that shows up in the dropdown
+  component?: FunctionComponent<VisProps> | ComponentClass<VisProps> // the view component for rendering the interface
+  disabled?: boolean // if you should show it or not
+  featureFlag?: string // designates a flag that should enable the panel type
```

Comment:
can you add a little note around `disabled` and `featureFlag` explaining that `disabled` takes precedence please?